### PR TITLE
Check LUA build type during world creating.

### PR DESCRIPTION
**Pull request by VlasovVitaly**
It's pretty simple.
Allow using lua mods only on lua builds.
Todo: control save loading, gettextify strings, doxygen.

### CODE REVIEW

File: src/mod_manager_ui.cpp [hunk 1]
```diff
@@ -130,6 +130,15 @@ std::string mod_ui::get_information(MOD_INFORMATION *mod)
         info << note;
     }
 
+#ifndef LUA
+    if ( mod->need_lua ) {
+        std::string lua_msg = "";
+        lua_msg += "<color_red>";
+        lua_msg += "This mod require lua but your CDDA build not support it!";
```
Comment (kevingranade) [inline src/mod_manager_ui.cpp]: requires

**Comment by kevingranade** [comment]
Good idea!

**Comment by BevapDin** [comment]
Additional to this, you could check for the presence of `mod->path + "/preload.lua"` or `mod->path + "/main.lua"`. If either exists, the mod requires Lua. Those files are loaded (`game::load_data_from_dir`) with the mod (if they exist). This should auto-detect Lua in most mods, the explicit setting would only be required when Lua is used in mapgen.

Btw. according to the site, http://www.lua.org/about.html, it's written with an uppercase L.

**Comment by VlasovVitaly** [comment]
Good. May it's better to scan for all *.lua files in ```mod->path```? I remembrer filesystem.cpp have such function. Anyway this will remove obsolete field from modinfo.json.

**Comment by BevapDin** [comment]
> Good. May it's better to scan for all *.lua files in  mod->path ? I remembrer filesystem.cpp have such function. Anyway this will remove obsolete field from modinfo.json.

Scanning the whole folder is not needed, if there is no preload.lua and no main.lua, no Lua script will be loaded at all.

>> the explicit setting would only be required when Lua is used in mapgen.

> omg, Is example of this exist?

As far as I know, it's not used anywhere, but it's easily possible: add a "lua" entry to the JSON object that already contains the "rows" or the "terrain" entry, like this:
```JSON
"object": {
    "rows":[
        "...",
    ],
    "lua": [ "game.add_msg(\"called lua mapgen\")" ],
    "terrain": {
        ".": "t_grass",
        "_": "t_pavement",
        ",": "t_pavement_y",
```

I think the explicit setting is fine as a fallback, when there are no Lua files.

**Comment by VlasovVitaly** [comment]
Done!

**Comment by VlasovVitaly** [comment]
May be need spelling check
```
This mod requires Lua but your CDDA build not support it!
Can't start in world [%s]. Some of mods require Lua support.
Can't add mod. This mod require Lua support.
Mod '%s' require Lua support.
```

**Comment by illi-kun** [comment]
"require"-> "requires" in 3rd & 4th sentences (like it's done in 1st one).

**Comment by illi-kun** [comment]
"require"-> "requires" in 3rd & 4th sentences (like it's done in 1st one).

**Comment by illi-kun** [comment]
"require"-> "requires" in 3rd & 4th sentences (like it's done in 1st one).

**Comment by illi-kun** [comment]
Yay, sorry for triple post due to bad connection.

**Pull request merged**